src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -220,27 +242,61 @@ def add_project_to_lpq(self, project_id: int) -> bool:
         This registers an intent to redirect all symbolication events triggered by the specified
         project to be redirected to the low priority queue.
 
-        This may throw an exception if there is some sort of issue registering the project with the
-        queue.
+        Applies a backoff timer to the project which prevents it from being automatically evicted
+        from the queue while that timer is active.
+
+        Returns True if the project was a new addition to the list. Returns False if it was already
+        assigned to the low priority queue. This may throw an exception if there is some sort of
+        issue registering the project with the queue.
         """
 
-        # This returns 0 if project_id was already in the set, 1 if it was added, and throws an
-        # exception if there's a problem. If this successfully completes then the project is
-        # expected to be in the set.
-        return int(self.cluster.sadd(LPQ_MEMBERS_KEY, project_id)) > 0
+        # If this successfully completes then the project is expected to be in the set.
+        was_added = int(self.cluster.sadd(LPQ_MEMBERS_KEY, project_id)) > 0
+        self._register_backoffs({project_id})
+        return was_added
```

Comment:
I would maybe recommend making it a concern of the caller because:
* less magic and surprises. the call does exactly what it says it does.
* the backoff should be a business requirement, so it should be exposed prominently?
* perf concerns that you mentioned.